test/integration/bastion/bastion_test.go

```diff
@@ -84,107 +84,111 @@ func validateFlags() {
-					filepath.Join(repoRoot, "example", "20-crd-extensions.gardener.cloud_clusters.yaml"),
-				},
+	By("starting test environment")
+	testEnv = &envtest.Environment{
+		UseExistingCluster: pointer.BoolPtr(true),
+		CRDInstallOptions: envtest.CRDInstallOptions{
+			Paths: []string{
+				filepath.Join(repoRoot, "example", "20-crd-extensions.gardener.cloud_bastions.yaml"),
+				filepath.Join(repoRoot, "example", "20-crd-extensions.gardener.cloud_clusters.yaml"),
 			},
-		}
-
-		cfg, err := testEnv.Start()
-		Expect(err).ToNot(HaveOccurred())
-		Expect(cfg).ToNot(BeNil())
+		},
+	}
 
-		By("setup manager")
-		mgr, err := manager.New(cfg, manager.Options{
-			MetricsBindAddress: "0",
-		})
-		Expect(err).ToNot(HaveOccurred())
+	cfg, err := testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cfg).ToNot(BeNil())
 
-		Expect(extensionsv1alpha1.AddToScheme(mgr.GetScheme())).To(Succeed())
-		Expect(awsinstall.AddToScheme(mgr.GetScheme())).To(Succeed())
+	By("setup manager")
+	mgr, err := manager.New(cfg, manager.Options{
+		MetricsBindAddress: "0",
+	})
+	Expect(err).ToNot(HaveOccurred())
 
-		Expect(bastionctrl.AddToManager(mgr)).To(Succeed())
+	Expect(extensionsv1alpha1.AddToScheme(mgr.GetScheme())).To(Succeed())
+	Expect(awsinstall.AddToScheme(mgr.GetScheme())).To(Succeed())
 
-		var mgrContext context.Context
-		mgrContext, mgrCancel = context.WithCancel(ctx)
+	Expect(bastionctrl.AddToManager(mgr)).To(Succeed())
 
-		By("start manager")
-		go func() {
-			err := mgr.Start(mgrContext)
-			Expect(err).NotTo(HaveOccurred())
-		}()
+	var mgrContext context.Context
+	mgrContext, mgrCancel = context.WithCancel(ctx)
 
-		// test client should be uncached and independent from the tested manager
-		c, err = client.New(cfg, client.Options{
-			Scheme: mgr.GetScheme(),
-			Mapper: mgr.GetRESTMapper(),
-		})
+	By("start manager")
+	go func() {
+		err := mgr.Start(mgrContext)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(c).NotTo(BeNil())
+	}()
 
-		flag.Parse()
-		validateFlags()
+	// test client should be uncached and independent from the tested manager
+	c, err = client.New(cfg, client.Options{
+		Scheme: mgr.GetScheme(),
+		Mapper: mgr.GetRESTMapper(),
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(c).NotTo(BeNil())
 
-		awsClient, err = awsclient.NewClient(*accessKeyID, *secretAccessKey, *region)
-		Expect(err).NotTo(HaveOccurred())
+	flag.Parse()
+	validateFlags()
 
-		amiID := determineBastionImage(ctx, awsClient)
-		extensionscluster, corecluster = newCluster(namespaceName, amiID)
-	})
+	awsClient, err = awsclient.NewClient(*accessKeyID, *secretAccessKey, *region)
+	Expect(err).NotTo(HaveOccurred())
 
-	AfterSuite(func() {
-		defer func() {
-			By("stopping manager")
-			mgrCancel()
-		}()
+	amiID := determineBastionImage(ctx, awsClient)
+	extensionscluster, corecluster = newCluster(namespaceName, amiID)
+})
 
-		By("running cleanup actions")
-		framework.RunCleanupActions()
+var _ = AfterSuite(func() {
+	defer func() {
```

Comment:
You could use `DeferCleanup` https://onsi.github.io/ginkgo/MIGRATING_TO_V2#defercleanup

Reply:
done with `5dbbbfc`